### DISCUSSION TITLE
[Feature Proposal]: Nested Environment Variables

### DISCUSSION THREAD

**Discussion by SentientMatt** [hoppscotch/hoppscotch#2109] [Ideas]
It would be great if we were able to used nested environment variable names so as to be able to dynamically build up an environment variable name to be resolved.

This has many uses, but in particular it would be a very useful in providing similar functionality to nested environments, while this is being figured out and/or considered.

Example.
**dev environment:**
env: dev
**test environment:**
env: test

**global environment**
dev_var: dev-value
test_var: test-value

**Usage**
<<<<**env**>>_var>>

This would then provide dev-value for dev and test-value for test.

While this example is fairly contrived, this become very useful where there are several dev and/or test profiles that might have different oauth credentials etc. but share the same API endpoints.

I have a basic PR ready to go if this is agreed to

**Comment by liyasthomas** [comment]
This has been implemented in https://github.com/hoppscotch/hoppscotch/commit/7fc757d79469d7bb72d8737aa87387fc693e4ac5.

Share what you think of Hoppscotch and how we can improve that experience in [❤️ Kind Words](https://github.com/hoppscotch/hoppscotch/discussions/categories/kind-words)

**Reply by PascalBru** [reply]
Hey @liyasthomas could you please explain how you insert the variables. I can't get it to work :(
